internal/hooks/auth_hooks.go

```diff
@@ -25,6 +27,78 @@ type HookOutput interface {
 	Error() string
 }
 
+// #nosec
+const MinimumViableTokenSchema = `{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "aud": {
+      "type": "string"
+    },
+    "exp": {
+      "type": "integer"
+    },
+    "jti": {
+      "type": "string"
+    },
+    "iat": {
+      "type": "integer"
+    },
+    "iss": {
+      "type": "string"
+    },
+    "nbf": {
+      "type": "integer"
+    },
+    "sub": {
+      "type": "string"
+    },
+    "email": {
+      "type": "string"
+    },
+    "phone": {
+      "type": "string"
+    },
+    "app_metadata": {
+      "type": "object",
+      "additionalProperties": true
+    },
+    "user_metadata": {
+      "type": "object",
+      "additionalProperties": true
+    },
+    "role": {
+      "type": "string"
+    },
+    "aal": {
+      "type": "string"
+    },
+    "amr": {
+      "type": "array",
+      "items": {
+        "type": "object"
+      }
+    },
+    "session_id": {
+      "type": "string"
+    }
+  },
+  "required": ["aud", "exp", "iat", "sub", "email", "phone", "role", "aal"]
+}`
+
+// GoTrueClaims is a struct thats used for JWT claims
+type GoTrueClaims struct {
```

Comment:
We introduce `GoTrueClaims` in `auth_hook` to avoid an import loop (this is in the `hooks` package which is imported by `api` so we cannot import `api.GoTrueClaims`. 

Also possible to move to a separate package.

Reply:
yeah would prefer to move it to a separate package in another PR so that we don't have to worry about them going out of sync